mitmproxy/net/encoding.py

```diff
@@ -174,12 +174,7 @@ def decode_zstd(content: bytes) -> bytes:
     if not content:
         return b""
     zstd_ctx = zstd.ZstdDecompressor()
-    try:
-        return zstd_ctx.decompress(content)
-    except zstd.ZstdError:
-        # If the zstd stream is streamed without a size header,
-        # try decoding with a 10MiB output buffer
-        return zstd_ctx.decompress(content, max_output_size=10 * 2**20)
+    return zstd_ctx.stream_reader(BytesIO(content)).read()
```

Comment:
Why does this work without passing `read_across_frames=True`?

According to https://python-zstandard.readthedocs.io/en/stable/decompressor.html#zstandard.ZstdDecompressor.stream_reader, it's false by default. Is this a bug in the underlying library, or why are things working here?

Reply:
Fair question, I don't know. I can add explicitly `read_across_frames=True` and test to ensure it works. Fair point, I'd sleep better if the code would align with the documentation. Let me play with this a little, will come back.

Reply:
So, I tested it and it works both with explicit False and True as well. I tracked down where this flags comes into play during Decompress' stream_reader().[read()](https://github.com/indygreg/python-zstandard/blob/518ade454f41ffbad8b9b542f6938cd308653ba2/zstandard/backend_cffi.py#L3288 ), it's [here ](https://github.com/indygreg/python-zstandard/blob/518ade454f41ffbad8b9b542f6938cd308653ba2/zstandard/backend_cffi.py#L3282) and it's part of an OR condition:

```python
        # Emit data if there is data AND either:
        # a) output buffer is full (read amount is satisfied)
        # b) we're at end of a frame and not in frame spanning mode
        return out_buffer.pos and (
            out_buffer.pos == out_buffer.size
            or zresult == 0
            and not self._read_across_frames
        )
```

I'm not going to pretend that I fully understand this part, I have some assumptions but I'm not familiar with zstd to this depth to say anything here with certainty. Still, I added the `read_across_frames=True` to the PR as based on my tests it works as well and based on the documentation it makes sense to have it.